samples/bookinfo/src/productpage/productpage.py

```diff
@@ -182,17 +177,66 @@ def getForwardHeaders(request):
     if 'user' in session:
         headers['end-user'] = session['user']
 
-    incoming_headers = ['x-request-id', 'x-datadog-trace-id', 'x-datadog-parent-id', 'x-datadog-sampled']
-
-    # Add user-agent to headers manually
-    if 'user-agent' in request.headers:
-        headers['user-agent'] = request.headers.get('user-agent')
+    # Keep this in sync with the headers in details and reviews.
+    incoming_headers = [
+        # All applications should propagate x-request-id. This header is
+        # included in access log statements and is used for consistent trace
+        # sampling and log sampling decisions in Istio.
+        'x-request-id',
+
+        # Lightstep tracing header. Propagate this if you use lightstep tracing
+        # in Istio (see
+        # https://istio.io/latest/docs/tasks/observability/distributed-tracing/lightstep/)
+        # Note: this should probably be changed to use B3 or W3C TRACE_CONTEXT.
+        # Lightstep recommends using B3 or TRACE_CONTEXT and most application
+        # libraries from lightstep do not support x-ot-span-context.
+        'x-ot-span-context',
+
+        # Datadog tracing header. Propagate these headers if you use Datadog
+        # tracing.
+        'x-datadog-trace-id',
+        'x-datadog-parent-id',
+        'x-datadog-sampling-priority',
+
+        # W3C Trace Context. Compatible with OpenCensusAgent and Stackdriver Istio
+        # configurations.
+        'traceparent',
+        'tracestate',
+
+        # Cloud trace context. Compatible with OpenCensusAgent and Stackdriver Istio
+        # configurations.
+        'x-cloud-trace-context',
+
+        # Grpc binary trace context. Compatible with OpenCensusAgent nad
+        # Stackdriver Istio configurations.
+        'grpc-trace-bin',
+
+        # b3 trace headers. Compatible with Zipkin, OpenCensusAgent, and
+        # Stackdriver Istio configurations. Commented out since they are
+        # propagated by the OpenTracing tracer above.
+        # 'x-b3-traceid',
+        # 'x-b3-spanid',
+        # 'x-b3-parentspanid',
+        # 'x-b3-sampled',
+        # 'x-b3-flags',
+
+        # Application-specific headers to forward.
+        'end-user',
```

Comment:
I don't think this should be an incoming header here - it's extracted from the session header (line 178, above)

Reply:
ah, you are right. Thanks!

Reply:
Actually, maybe it's OK. If they are both set, the end-user header will override the session user, but that should never happen anyway.

Reply:
Removed, I think it's clearer not to double-handle it.